File: src/main/java/com/patres/kotlinplusplus/e_string/b_multi_line_string/java/MultiLineStringApp.java
package com.patres.kotlinplusplus.e_string.b_multi_line_string.java;

public class MultiLineStringApp {

    private static final String HELLO = "Hello\n" +
            "My friend\n" +
            "How are you";


    private static final String FIND_PAYMENT_AND_VENDOR_BY_ID =
                    "SELECT\n" +
                    "    PMT_ID, VENDOR_ID\n" +
                    "FROM\n" +
                    "    PAYMENT PM\n" +
                    "JOIN VENDOR V\n" +
                    "    ON P.VENDOR_ID = V.VENDOR_ID\n" +
                    "WHERE\n" +
                    "    P.PMT_ID = 3;";

    public static void main(String[] args) {
        System.out.println(HELLO);
    }

}

File: src/main/java/com/patres/kotlinplusplus/e_string/b_multi_line_string/kotlin/MultiLineStringApp.kt
package com.patres.kotlinplusplus.e_string.b_multi_line_string.kotlin

val FIND_PAYMENT_AND_VENDOR_BY_ID =
    """
    SELECT
        PMT_ID, VENDOR_ID
    FROM
        PAYMENT PM
    JOIN VENDOR V
        ON P.VENDOR_ID = V.VENDOR_ID
    WHERE
        P.PMT_ID = 3;
    """

private const val helloTrim = """Hello
        |My friend
        |How are you?
        """

fun main() {
    println(helloTrim.trimMargin())
}